src/Take.Elephant.Specialized/Cache/OnDemandCacheSetMap.cs

```diff
@@ -30,17 +32,32 @@ public override async Task<ISet<TValue>> GetValueOrDefaultAsync(TKey key, Cancel
                     cacheValue);
             }
 
-            var sourceValue = await Source.GetValueOrDefaultAsync(key, cancellationToken).ConfigureAwait(false);
-            if (sourceValue == null) return null;
+            var sourceValue = await ((ISetMap<TKey, TValue>)Source).GetValueOrDefaultAsync(key, cancellationToken).ConfigureAwait(false);
+
+            if (sourceValue is null)
+            {
+                if (((ISetMap<TKey, TValue>)Cache).SupportsEmptySets)
```

Comment:
You can replace with `SupportsEmptySets`.
If the intention is to show explicitly that the Cache should support this, maybe introducing a private member name like `CacheSupportsEmptySets` - to be used here and on `SupportsEmptySets` - could be a nice way to solve both cases and remove the duplicated check:

```csharp

public bool SupportsEmptySets => __cacheSupportsEmptySets;

...

            if (sourceValue is null)
            {
                if (_cacheSupportsEmptySets)

...

private bool _cacheSupportsEmptySets => ((ISetMap<TKey, TValue>)Cache).SupportsEmptySets;

Reply:
Nice catch :)